### ISSUE TITLE
Knob not updated on new data

### ISSUE THREAD

**Issue by Syeberman** [talpor/django-dashing#36]
`Graph` and `List` both update on `$.extend( this.scope, scope )`, however `Knob` (specifically, `scope.value` and `scope.data.fgColor`) does not.  (I want a knob that changes its foreground colour based on the value, in the standard "green/yellow/red" scheme.)

For example, the following will update the numerical value displayed (i.e. the one hidden by `scope.data.displayInput`), but not the arc itself:
```javascript
dashboard.addWidget( "MyWidget", "Knob", {
    getData: function() {
        var scope = GetWidgetScope();
        $.extend( this.scope, scope );
    }
});
```

I've been able to work around this by adding these `trigger` calls to the end of `getData`, but I'd rather this was handled by the library itself:
```javascript
$( ".dashing-knob" ).trigger( "change" );           // updates the value
$( ".dashing-knob" ).trigger( "configure", data );  // updates the colour/etc
```

**Comment by nitronick600** [comment]
I'm seeing a related issue with a custom number widget not updating the "updatedAt" and "moreInfo" fields.  I am updating the 'value' and 'title' values using the same method.  

I have a very basic project called EdgeDash (https://github.com/nitronick600/EdgeDash) for which you can see my current configuration.  I'm currently having issues with the WorkToClientWidget

**Comment by wimkerkhoff** [comment]
"updatedAt" is working for me... here is working example from my dashing-config.js

```
dashboard.addWidget('NagiosUnhandledHosts', 'KTINagios', {
  row: 1,
  col: 1,
  interval: 1000*10,
   getData: function () {
      var apple = this;
       $.get('/NagiosUnhandledHosts/', function(data) {
          apple.data.title = data.title;
          apple.data.moreInfo = data.moreInfo;
          apple.data.detail = data.detail;
          apple.data.updatedAt = data.updatedAt;
          apple.data.value = data.value;
      });
   }
});
```

**Comment by zubairazami** [comment]
get_value() was not there in the widgets.py

```
class KnobWidget(Widget):
    title = ''
    data = {}
    detail = ''
    more_info = ''
    updated_at = ''

    def get_title(self):
        return self.title

    def get_data(self):
        return self.data

    def get_detail(self):
        return self.detail

    def get_more_info(self):
        return self.more_info

    def get_updated_at(self):
        return self.updated_at

    def get_context(self):
        return {
            'title': self.get_title(),
            'data': self.get_data(),
            'detail': self.get_detail(),
            'moreInfo': self.get_more_info(),
            'updatedAt': self.get_updated_at(),
        }
```
after adding value, it works well.

```
class KnobWidget(Widget):
    title = ''
    data = {}
    detail = ''
    more_info = ''
    updated_at = ''
    value = ''

    def get_value(self):
        return self.value

    def get_title(self):
        return self.title

    def get_data(self):
        return self.data

    def get_detail(self):
        return self.detail

    def get_more_info(self):
        return self.more_info

    def get_updated_at(self):
        return self.updated_at

    def get_context(self):
        return {
            'title': self.get_title(),
            'value': self.get_value(),
            'data': self.get_data(),
            'detail': self.get_detail(),
            'moreInfo': self.get_more_info(),
            'updatedAt': self.get_updated_at(),
        }
```

**Comment by individuo7** [comment]
I just released the version 0.3.1, this should be resolved there
